packages/commands/src/bitcoin/depositAndCall.ts

```diff
@@ -0,0 +1,158 @@
+import confirm from "@inquirer/confirm";
+import axios from "axios";
+import * as bitcoin from "bitcoinjs-lib";
+import { Command } from "commander";
+import ECPairFactory from "ecpair";
+import { ethers } from "ethers";
+import * as ecc from "tiny-secp256k1";
+
+import { BITCOIN_FEES } from "../../../../types/bitcoin.constants";
+import type { BtcUtxo } from "../../../../types/bitcoin.types";
+import {
+  makeCommitTransaction,
+  makeRevealTransaction,
+  SIGNET,
+} from "../../../../utils/bitcoin.helpers";
+import {
+  bitcoinEncode,
+  EncodingFormat,
+  OpCode,
+  trimOx,
+} from "../../../client/src/bitcoinEncode";
+
+// Initialize Bitcoin library with ECC implementation
+bitcoin.initEccLib(ecc);
+
+/**
+ * Options for the deposit-and-call command
+ */
+interface depositAndCallOptions {
+  amount: string;
+  api: string;
+  gateway: string;
+  privateKey: string;
+  receiver: string;
+  revertAddress: string;
+  types: string[];
+  values: string[];
+}
+
+/**
+ * Main function that executes the deposit-and-call operation.
+ * Creates and broadcasts both commit and reveal transactions to perform a cross-chain call.
+ *
+ * @param options - Command options including amounts, addresses, and contract parameters
+ */
+const main = async (options: depositAndCallOptions) => {
+  // Set up Bitcoin key pair
+  const ECPair = ECPairFactory(ecc);
+  const pk = options.privateKey;
+  if (!pk) throw new Error("missing private key");
```

Comment:
We should check against a zod schema like the rest of the commands.

Reply:
15bdb4e